src/libs/ReportUtils.js

```diff
@@ -1916,13 +1915,6 @@ function shouldReportBeInOptionList(report, currentReportId, isInGSDMode, iouRep
         return false;
     }
 
-    // Include the currently viewed report. If we excluded the currently viewed report, then there
```

Comment:
Are you sure we can just remove this condition and it will work correctly?

Reply:
Haha okay, I may spoke too fast. I see that some tests are failing, indicating the need for this code. Will double and triple check 😊

Reply:
Yeah, we can't remove that, otherwise the sorting in GSD/focus mode would break. I added it back but changed the general structure a bit to still be optimised.